ramips: add support for Wavlink WL-WN570HA2

jonathan-sturges opened the pull request:
ramips: add support for Wavlink WL-WN570HA2

The Wavlink Aerial HD3 (WL-WN570HA2) is an AC1200 dual-band outdoor
access point.  It supports 802.3AT/AF PoE and is IP67 waterproof. It is based on the MediaTek MT7628DAN SoC and MT7613BEN WiFi 5 chip. This model uses the 100Mbit LAN and 2.4Ghz WiFi elements of the MT7628 and the 5Ghz WiFi of the MT7613.

Specification:
- SoC: MediaTek MT7628DAN (1C/1T) @ 580MHz
- RAM: 64MB DDR2 (integrated in SoC)
- FLASH: 16MB SPI NOR (Fudan FM25Q128A)
- Ethernet: 1x 10/100Mbps
- WiFi: 2.4/5 GHz 2T2R
  - 2.4GHz MediaTek MT7628DAN bgn
  - 5GHz MediaTek MT7613BEN nac
- Antennas: 2x detachable, dual-band 7dBi with RP-SMA connectors.
- USB: none
- BTN: Reset
- LED: 6 total: power; WAN/LAN; WiFi; WiFi low; WiFi med; Wifi high
- UART:  surface-mount on PCB.  Pins are marked via silkscreen. pin1 (square pad, towards Ethernet)=Vcc, pin2=RX, pin3=TX, pin4=GND.  Settings: 57600/8N1. NOTE: The TX & RX silkscreens were reversed on my test unit.

Installation:
1) This device requires a HTTP recovery procedure to do an initial load
  of OpenWRT.  You will need:
  a. A web browser (private window recommended)
  b. Configure an Ethernet interface to 192.168.1.x/24; don't use .1
  c. Connect a cable between the computer and the Wavlink's PoE injector.
2) Put the Wavlink in HTTP recovery mode.
  a. Do this by pressing and holding the reset button on the bottom while
  powering the unit on.
  b. As soon as all 6 LEDs light up blue (roughly 2-3 seconds), release
  the button.
  c. The LEDs should all remain lit, indicating it's in HTTP recovery.
3) Point the browser at http://192.168.1.1/index.html 4) Click "Choose File" and select the OpenWRT sysupgrade image. 5) Click the "Update Firmware" button and wait while the unit flashes the image and reboots.
6) When the system comes back up fully, only the power LED will be lit. Wait an extra minute then you should be able to reach OpenWRT on http://192.168.1.1
5) Log into LuCI as root; there is no password.

Revert to the OEM Firmware:
--------------------------
* U-boot HTTP: Follow the HTTP recovery steps, and use a firmware image downloaded from Wavlink.

Djfe commented on target/linux/ramips/dts/mt7628an_wavlink_wl-wn570ha2.dts:
```diff
@@ -0,0 +1,185 @@
+
+&pcie0 {
+	wifi@0,0 {
+		reg = <0x0000 0 0 0 0>;
```
you could add a `compatible` line before the `reg` line
https://github.com/openwrt/openwrt/blob/dfacae304915bcd316ac2d8c69bfe3f15749f5fa/target/linux/ramips/dts/mt7628an_cudy_tr1200-v1.dts#L175

Djfe commented:
Just a quick side note (nothing to be discouraged by but you should know about it): usability under OpenWrt might be not as great as with Wavlink's original firmware:
**mt76** currently does not support dfs event detection for the 5GHz chip mt7613, so you are limited to indoor channels 36-48. outdoor operation under law is limited to 2.4ghz in most cases.

Also 64MiB of ram is fairly limited, too (for a dual band AP. if you keep 5GHz disabled it should be fine for a few years and further OpenWrt updates)

atleast the flash chip is of a decent size :)

jonathan-sturges commented:
Thanks for the notes!
I've already seen the limits of the 64MB RAM and have resorted to CLI sysupgrades in testing.  Luckily the Uboot HTTP recovery accepts an OpenWRT sysupgrade image just in case.
Re: mt7613, should the UNII-3 channels be expected work, where allowed?

Djfe commented:
Looks all good to me.

you could add this entry to allow u-boot-env to be accessed easily
https://github.com/openwrt/openwrt/blob/main/package/boot/uboot-tools/uboot-envtools/files/ramips#L34

you would then add this package and could read your uboot environment from mtd1
https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/image/mt76x8.mk#L72

though your device doesn't require this but why not :)

Djfe commented:
I'm from Europe so I don't know much about UNII-3 here.
in Europe those are limited to short range usage (25mW) and not supported by most clients.

according to https://www.aami.org/medical-device-connectivity/medical-device-connectivity-faqs-standards-802-11#:~:text=For%20the%20U.S.%2C%20there%20are,and%205.73%20GHz%20frequency%20range.
> For the U.S., there are nine non-DFS channels: 36, 40, 44, 48, 149, 153, 157, 161, and 165.

so in the USA this might not be as problematic :)

namiltd commented on target/linux/ramips/dts/mt7628an_wavlink_wl-wn570ha2.dts:
```diff
@@ -0,0 +1,186 @@
+
+		wifi {
+			function = LED_FUNCTION_WLAN;
+			label = "blue:wifi";
```
remove this

jonathan-sturges replied:
ok, done.

namiltd commented on target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:
```diff
@@ -168,6 +168,14 @@ wavlink,wl-wn570ha1)
+	ucidef_set_led_rssi "wifi-low" "wifi-low" "blue:wifi-low" "phy1-ap0" "1" "100"
+	ucidef_set_led_rssi "wifi-med" "wifi-med" "blue:wifi-med" "phy1-ap0" "50" "100"
+	ucidef_set_led_rssi "wifi-high" "wifi-high" "blue:wifi-high" "phy1-ap0" "85" "100"
+	ucidef_set_led_wlan "wifi_led" "wifi" "blue:wifi" "phy0tpt"
```
ucidef_set_led_wlan "wifi_led" "wifi" "blue:wlan" "phy0tpt"

jonathan-sturges commented on target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:
```diff
@@ -168,6 +168,14 @@ wavlink,wl-wn570ha1)
+	ucidef_set_led_rssi "wifi-low" "wifi-low" "blue:wifi-low" "phy1-ap0" "1" "100"
+	ucidef_set_led_rssi "wifi-med" "wifi-med" "blue:wifi-med" "phy1-ap0" "50" "100"
+	ucidef_set_led_rssi "wifi-high" "wifi-high" "blue:wifi-high" "phy1-ap0" "85" "100"
+	ucidef_set_led_wlan "wifi_led" "wifi" "blue:wifi" "phy0tpt"
```
thanks, didn't know that could go in the DTS now; done!

hauke commented on target/linux/ramips/dts/mt7628an_wavlink_wl-wn570ha2.dts:
```diff
@@ -0,0 +1,186 @@
+
+		led_power: power {
+			function = LED_FUNCTION_POWER;
+			label = "blue:power";
```
The label property is not needed when you specify `function` and `color`.

hauke commented on target/linux/ramips/dts/mt7628an_wavlink_wl-wn570ha2.dts:
```diff
@@ -0,0 +1,186 @@
+		};
+
+		wifi-high {
+			label = "blue:wifi-high";
```
Please use this instead:
```
function = "wifi-high";
```

jonathan-sturges committed:
ramips: add support for Wavlink WL-WN570HA2

The Wavlink Aerial HD3 (WL-WN570HA2) is an AC1200 dual-band outdoor
access point.  It supports 802.3AT/AF PoE and is IP67 waterproof.
It is based on the MediaTek MT7628DAN SoC and MT7613BEN WiFi 5 chip.
This model uses the 100Mbit LAN and 2.4Ghz WiFi elements of the
MT7628 and the 5Ghz WiFi of the MT7613.

Specification:
- SoC: MediaTek MT7628DAN (1C/1T) @ 580MHz
- RAM: 64MB DDR2 (integrated in SoC)
- FLASH: 16MB SPI NOR (Fudan FM25Q128A)
- Ethernet: 1x 10/100Mbps
- WiFi: 2.4/5 GHz 2T2R
  - 2.4GHz MediaTek MT7628DAN bgn
  - 5GHz MediaTek MT7613BEN nac
- Antennas: 2x detachable, dual-band 7dBi with RP-SMA connectors.
- USB: none
- BTN: Reset
- LED: 6 total: power; WAN/LAN; WiFi; WiFi low; WiFi med; Wifi high
- UART:  surface-mount on PCB.  Pins are marked via silkscreen.
   pin1 (square pad, towards Ethernet)=Vcc, pin2=RX,
   pin3=TX, pin4=GND.  Settings: 57600/8N1.
   NOTE: The TX & RX silkscreens were reversed on my test unit.

Installation:
1) This device requires a HTTP recovery procedure to do an initial load
  of OpenWRT.  You will need:
  a. A web browser (private window recommended)
  b. Configure an Ethernet interface to 192.168.1.x/24; don't use .1
  c. Connect a cable between the computer and the Wavlink's PoE injector.
2) Put the Wavlink in HTTP recovery mode.
  a. Do this by pressing and holding the reset button on the bottom while
  powering the unit on.
  b. As soon as all 6 LEDs light up blue (roughly 2-3 seconds), release
  the button.
  c. The LEDs should all remain lit, indicating it's in HTTP recovery.
3) Point the browser at http://192.168.1.1/index.html
4) Click "Choose File" and select the OpenWRT sysupgrade image.
5) Click the "Update Firmware" button and wait while the unit flashes
the image and reboots.
6) When the system comes back up fully, only the power LED will be lit.
Wait an extra minute then you should be able to reach OpenWRT on
http://192.168.1.1
5) Log into LuCI as root; there is no password.

Revert to the OEM Firmware:
--------------------------
* U-boot HTTP:
  Follow the HTTP recovery steps, and use a firmware image downloaded
  from Wavlink.

Signed-off-by: Jonathan Sturges <jsturges@redhat.com>
Link: https://github.com/openwrt/openwrt/pull/18856
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

jonathan-sturges commented:
The main things are, make sure you set your country, adjust the WiFi power, and avoid the DFS channels.  Hopefully that helps.
